synapse/event_auth.py

```diff
@@ -1075,8 +1080,6 @@ def _verify_third_party_invite(
 
     if signed["mxid"] != event.state_key:
         return False
-    if signed["token"] != token:
-        return False
```

Comment:
On line 1064 we set `token = signed["token"]`. Neither `token` nor `signed` change in the interim, so this is now just checking against itself.

Reply:
:+1: I dug into the past to see if it was different and this just seems to be the `third_party_invite` content shape evolving and mistaken refactors from 8 years ago.